### PR TITLE
Refactor model card init

### PR DESCRIPTION
**Pull request by BenjaminBossan**
_This PR is just a basis for discussion_

### CODE REVIEW

File: skops/card/_model_card.py [hunk 1]
```diff
@@ -504,9 +505,10 @@ def __init__(
         self._data: dict[str, Section] = {}
         self._metrics: dict[str, str | float | int] = {}
 
-        self._populate_template(model_diagram=model_diagram)
-
-    def _populate_template(self, model_diagram: bool | Literal["auto"] | str):
+    def populate_template(
```
Comment (adrinjalali) [inline skops/card/_model_card.py:508]: users shouldn't have to call this one, do they?

**Review by adrinjalali — COMMENTED** [review]
I think we can remove the `        if self.template == Templates.skops.value:` from this method.

Also, we can remove `            if self.template == Templates.skops.value:` like statements from the other methods.

File: skops/card/_model_card.py (hunk 1 above)
Comment (BenjaminBossan) [reply skops/card/_model_card.py:508]: If they didn't, then the card would always be empty, would it not?

**Review by BenjaminBossan — COMMENTED** [review]
> I think we can remove the `if self.template == Templates.skops.value`: from this method.

You mean that the code below that should be called unconditionally? This would require there to be defaults for the section names in case a non-default template is being used.

> Also, we can remove `if self.template == Templates.skops.value:` like statements from the other methods.

If we set default section names, we could do that. However, we could do that regardless of the change in this PR.

File: skops/card/_model_card.py (hunk 1 above)
Comment (adrinjalali) [reply skops/card/_model_card.py:508]: when I create a card based on a template, I expect at least the default sections to be there.

**Comment by adrinjalali** [comment]
> This would require there to be defaults for the section names in case a non-default template is being used.

Yes.

File: skops/card/_model_card.py (hunk 1 above)
Comment (BenjaminBossan) [reply skops/card/_model_card.py:508]: Okay, but then it's not different from what we already have. I think this means that we could add the default section names to the `add_` methods but that's it. This PR could be closed.

Comment (adrinjalali) [reply skops/card/_model_card.py:508]: looks like it, and simplify their code.

**Comment by BenjaminBossan** [comment]
Conclusion: Not a worthy change, but we can set default section names for the `add_` methods, which would allow to get rid of the template checks inside of them, resulting in simplified code.

**Pull request closed without merging**